core/src/test/scala/unit/kafka/coordinator/GroupMetadataManagerTest.scala

```diff
@@ -86,6 +88,263 @@ class GroupMetadataManagerTest {
   }
 
   @Test
+  def testLoadOffsetsWithoutGroup() {
+    val offsetTopicPartition = new TopicPartition(Topic.GroupMetadataTopicName, groupPartitionId)
+    val logMock =  EasyMock.mock(classOf[Log])
+    val fileRecordsMock = EasyMock.mock(classOf[FileRecords])
+    val startOffset = 15L
+    val endOffset = 18L
+
+    val committedOffsets = Map(
+      new TopicPartition("foo", 0) -> 23L,
+      new TopicPartition("foo", 1) -> 455L,
+      new TopicPartition("bar", 0) -> 8992L
+    )
+    val offsetCommitRecords = committedOffsets.map { case (topicPartition, offset) =>
+      val offsetAndMetadata = OffsetAndMetadata(offset)
+      val offsetCommitKey = GroupMetadataManager.offsetCommitKey(groupId, topicPartition)
+      val offsetCommitValue = GroupMetadataManager.offsetCommitValue(offsetAndMetadata)
+      Record.create(offsetCommitKey, offsetCommitValue)
+    }.toSeq
+
+    val records = MemoryRecords.withRecords(startOffset, offsetCommitRecords: _*)
+
+    EasyMock.expect(replicaManager.getLog(offsetTopicPartition)).andStubReturn(Some(logMock))
+    EasyMock.expect(logMock.logStartOffset).andStubReturn(startOffset)
+    EasyMock.expect(replicaManager.getHighWatermark(offsetTopicPartition)).andStubReturn(Some(endOffset))
+    EasyMock.expect(logMock.read(EasyMock.eq(startOffset), EasyMock.anyInt(), EasyMock.eq(None), EasyMock.eq(true)))
+      .andReturn(FetchDataInfo(LogOffsetMetadata(startOffset), fileRecordsMock))
+    EasyMock.expect(fileRecordsMock.readInto(EasyMock.anyObject(classOf[ByteBuffer]), EasyMock.anyInt()))
+      .andReturn(records.buffer)
+
+    EasyMock.replay(logMock, replicaManager, fileRecordsMock)
+    
+    groupMetadataManager.loadGroupsAndOffsets(offsetTopicPartition, _ => ())
+
+    val loadedGroup = groupMetadataManager.getGroup(groupId)
```

Comment:
Nit: this could be written as `val group = groupMetadataManager.getGroup(groupId).getOrElse(fail("..."))` and then you won't need to pattern match. Similar for other tests that do the same.